File: backend/src/CoreMeet.Api/Domain/Entities/ChatMessage.cs
namespace CoreMeet.Api.Domain.Entities;

public class ChatMessage
{
    public Guid Id { get; set; } = Guid.NewGuid();

    public Guid MeetingId { get; set; }

    public Meeting? Meeting { get; set; }

    public Guid SenderParticipantId { get; set; }

    public string SenderName { get; set; } = string.Empty;

    public string Content { get; set; } = string.Empty;

    public DateTime SentAt { get; set; } = DateTime.UtcNow;
}

File: mobile/lib/src/meetings/models/chat_message.dart
import 'package:flutter/foundation.dart';

import 'meeting.dart' show parseApiTime;

@immutable
class ChatMessage {
  const ChatMessage({
    required this.id,
    required this.senderParticipantId,
    required this.senderName,
    required this.content,
    required this.sentAt,
  });

  final String id;
  final String senderParticipantId;
  final String senderName;
  final String content;
  final DateTime sentAt;

  factory ChatMessage.fromJson(Map<String, dynamic> j) => ChatMessage(
        id: '${j['id']}',
        senderParticipantId: '${j['senderParticipantId']}',
        senderName: j['senderName'] as String? ?? '',
        content: j['content'] as String? ?? '',
        sentAt: parseApiTime(j['sentAt']),
      );
}
